tests/six_test.py

```diff
@@ -382,6 +382,26 @@ def test_fix_six_noop(s):
             'print(next(iter({1:2}.values())))\n',
             id='six.itervalues inside next(...)',
         ),
+        pytest.param(
+            'six.itervalues({} or {})\n',
+            '({} or {}).values()\n',
+            id='six.itervalues with boolean logic inside call',
+        ),
+        pytest.param(
+            'six.iterkeys({} if True else {})\n',
+            '({} if True else {}).keys()\n',
+            id='six.itervalues with if-expression inside call',
+        ),
+        pytest.param(
+            'six.iteritems({} for _ in "")\n',
+            '({} for _ in "").items()\n',
+            id='six.itervalues with generator expression inside call',
+        ),
+        pytest.param(
+            'six.int2byte(1 & 1)\n',
```

Comment:
int2byte is one of the few that *doesn't* need parens added

Reply:
Well, I just added this test for your mentioned `BinOp` - because my imagination is not big enough to come up wit a different one, beside a 3.9 only dict merge.

Reply:
that's fine, the other examples aren't valid calls either (itervalues on a generator for example) -- feel free to use a merge operator

but also keep this test and fix the output since it's adding an extra set of parens incorrectly currently

Reply:
Done. But regarding the generator: I only came up with one idea - which is valid in Python3 but not in six for Python2. Any other suggestion?